Description: Add a quiet mode to gptsync for non-interactive use
 gptsync's output is a nuisance when used in scripts.
Author: Julien BLACHE <jblache@debian.org>

Index: refit-0.14/gptsync/Makefile.unix
===================================================================
--- refit-0.14.orig/gptsync/Makefile.unix	2010-04-19 18:33:15.842932277 +0200
+++ refit-0.14/gptsync/Makefile.unix	2010-04-19 18:33:14.643931533 +0200
@@ -44,7 +44,7 @@
 gptsync.unix.o: gptsync.c
 	$(CC) $(CPPFLAGS) $(CFLAGS) -c $< -o $@
 os_unix.gptsync.o: os_unix.c
-	$(CC) $(CPPFLAGS) -DPROGNAME=gptsync $(CFLAGS) -c $< -o $@
+	$(CC) $(CPPFLAGS) -DPROGNAME=gptsync -DGPTSYNC $(CFLAGS) -c $< -o $@
 
 $(SHOWPART_TARGET): $(SHOWPART_OBJS)
 	$(CC) $(LDFLAGS) -o $@ $(SHOWPART_OBJS) $(LIBS)
@@ -52,7 +52,7 @@
 showpart.unix.o: showpart.c
 	$(CC) $(CPPFLAGS) $(CFLAGS) -c $< -o $@
 os_unix.showpart.o: os_unix.c
-	$(CC) $(CPPFLAGS) -DPROGNAME=showpart -DNOREADONLYWARN $(CFLAGS) -c $< -o $@
+	$(CC) $(CPPFLAGS) -DPROGNAME=showpart -DSHOWPART -DNOREADONLYWARN $(CFLAGS) -c $< -o $@
 
 lib.unix.o: lib.c
 	$(CC) $(CPPFLAGS) $(CFLAGS) -c $< -o $@
Index: refit-0.14/gptsync/gptsync.8
===================================================================
--- refit-0.14.orig/gptsync/gptsync.8	2010-04-19 18:33:15.808931911 +0200
+++ refit-0.14/gptsync/gptsync.8	2010-04-19 18:33:14.643931533 +0200
@@ -1,16 +1,24 @@
-.TH "gptsync" 8 "2006 Jul 2" "Debian" "rEFIt"
+.TH "gptsync" 8 "2008 May 23" "Debian" "rEFIt"
 .SH NAME
 gptsync \- GPT partition table to MBR partition table synchronisation
-.SH "SYNOPSIS"
-.BI "gptsync " "device"
-.SH "DESCRIPTION"
+
+.SH SYNOPSIS
+.B gptsync
+[\fI\-q\fR] \fIdevice\fR
+
+.SH DESCRIPTION
 Reads the GPT partition table on the device and synchronise content of
 MBR partition table on the device.  Useful for situations (as in
 mactel linux) where legacy operating systems require MBR partition
 table to function properly, while most other operating systems can
 work with GPT.
 
-.SH "Author"
+.SH OPTIONS
+.TP
+\fB\-q\fR
+Do not print any message, do not prompt to modify the MBR.
+
+.SH AUTHOR
 Written by Christoph Pfisterer. This manual page contributed for Debian by
 Junichi Uekawa <dancer@debian.org>, but may be used for others.
 
Index: refit-0.14/gptsync/os_unix.c
===================================================================
--- refit-0.14.orig/gptsync/os_unix.c	2010-04-19 18:33:15.891931901 +0200
+++ refit-0.14/gptsync/os_unix.c	2010-04-19 18:33:14.643931533 +0200
@@ -43,8 +43,8 @@
 #define PROGNAME_S STRINGIFY2(PROGNAME)
 
 // variables
-
 static int      fd;
+int quiet = 0;
 
 //
 // error functions
@@ -135,7 +135,13 @@
 UINTN input_boolean(CHARN *prompt, BOOLEAN *bool_out)
 {
     int c;
-    
+
+    if (quiet)
+      {
+	*bool_out = TRUE;
+	return 0;
+      }
+
     printf("%s", prompt);
     fflush(NULL);
     
@@ -168,7 +174,10 @@
     char formatbuf[256];
     char buf[4096];
     int i;
-    
+
+    if (quiet)
+      return;
+
     for (i = 0; format[i]; i++)
         formatbuf[i] = (format[i] > 255) ? '?' : (char)(format[i] & 0xff);
     formatbuf[i] = 0;
@@ -180,6 +189,18 @@
     printf("%s", buf);
 }
 
+void usage(void)
+{
+#ifdef GPTSYNC
+  fprintf(stderr, "Usage: " PROGNAME_S " [-q] <device>\n");
+  fprintf(stderr, "\t-q\tquiet mode\n");
+
+#elif defined(SHOWPART)
+
+  fprintf(stderr, "Usage: " PROGNAME_S " <device>\n");
+#endif
+}
+
 //
 // main entry point
 //
@@ -192,14 +213,37 @@
     UINT64      filesize;
     char        *reason;
     int         status;
-    
+
     // argument check
+#ifdef GPTSYNC
+    switch (argc)
+      {
+        case 2:
+	  filename = argv[1];
+	  break;
+        case 3:
+	  if (strncmp("-q", argv[1], 2) != 0)
+	    {
+	      usage();
+	      return 1;
+	    }
+	  quiet = 1;
+	  filename = argv[2];
+	  break;
+        default:
+	  usage();
+	  return 1;
+      }
+
+#elif defined(SHOWPART)
+
     if (argc != 2) {
-        fprintf(stderr, "Usage: " PROGNAME_S " <device>\n");
-        return 1;
+	usage();
+	return 1;
     }
     filename = argv[1];
-    
+#endif
+
     // set input to unbuffered
     fflush(NULL);
     setvbuf(stdin, NULL, _IONBF, 0);
